### PR TITLE
Use telemetry and client funcs from shared module

### PR DESCRIPTION
**Pull request by alexhung**

### CODE REVIEW

File: pkg/xray/provider_test.go [hunk 1]
```diff
@@ -29,15 +30,25 @@ func TestProvider(t *testing.T) {
 }
 
 func getTestResty(t *testing.T) *resty.Client {
-	if v := os.Getenv("JFROG_URL"); v == "" {
-		t.Error("JFROG_URL must be set for acceptance tests")
+	var ok bool
+	var xrayUrl string
+	if xrayUrl, ok = os.LookupEnv("XRAY_URL"); !ok {
```
Comment (alexhung) [inline pkg/xray/provider_test.go:35]: Added these for now to support both env vars. Need to move `GetEnvVarWithFallback ` from https://github.com/jfrog/terraform-provider-artifactory/blob/master/pkg/acctest/test.go#L75 to the shared module first.

File: pkg/xray/provider_test.go [hunk 2]
```diff
@@ -29,15 +30,25 @@ func TestProvider(t *testing.T) {
...
-	restyClient, err := buildResty(os.Getenv("JFROG_URL"))
+	restyClient, err := client.Build(xrayUrl, "")
 	if err != nil {
 		t.Error(err)
 	}
-	accessToken := os.Getenv("XRAY_ACCESS_TOKEN")
-	restyClient, err = addAuthToResty(restyClient, accessToken)
+
+	var accessToken string
+	if accessToken, ok = os.LookupEnv("XRAY_ACCESS_TOKEN"); !ok {
```
Comment (alexhung) [inline pkg/xray/provider_test.go:46]: Same here to support both env vars

File: GNUmakefile [hunk 3]
```diff
@@ -46,7 +46,7 @@ attach:
 
 acceptance: fmtcheck
 	export TF_ACC=true && \
-	go test -ldflags="-X '${PKG_VERSION_PATH}.Version=${NEXT_VERSION}-test'" -v -parallel 20 -timeout 20m ./pkg/...
+	go test -ldflags="-X '${PKG_VERSION_PATH}.Version=${NEXT_VERSION}'" -v -parallel 20 -timeout 20m ./pkg/...
```
Comment (danielmkn) [inline GNUmakefile:49]: why did you remove `-test`? It might create a noise in BI data.

Comment (alexhung) [reply GNUmakefile:49]: Ok. Fair enough.

Comment (danielmkn) [reply GNUmakefile]: I've noticed same in RT provider and in projects. A bit too late, but we need to remove it as well please.

**Review by danielmkn — APPROVED** [review]
